Source/Core/Core/DSP/Jit/x64/DSPJitArithmetic.cpp

```diff
@@ -82,7 +84,7 @@ void DSPEmitter::andcf(const UDSPInstruction opc)
     OR(16, sr_reg, Imm16(SR_LOGIC_ZERO));
     FixupBranch exit = J();
     SetJumpTarget(notLogicZero);
-    AND(16, sr_reg, Imm16(~SR_LOGIC_ZERO));
+    AND(16, sr_reg, Imm16(~SR_LOGIC_ZERO & 0xFFFF));
```

Comment:
Ugh, my bad - maybe I'll even enable this warning, because it looks fairly useful. That said, is there absolutely no way to fix it without any typecasting? Function style cast looks "nice" here, but it's still a cast.

Reply:
In this case, type casting would be necessary.

Reply:
Type casting like you suggested generates warning `C4310 - cast truncates constant value`. Maybe the way I have it is preferable after all.

Reply:
Given how even Microsoft headers generate C4365, I would just ignore that fact.

Reply:
Maybe we should disable C4310? IMO an explicit cast is clearer.

Reply:
I'm not sure. C4310 sounds somewhat useful.